packages/aws-cdk-lib/aws-dynamodb/lib/global-table.ts

```diff
@@ -0,0 +1,2152 @@
+            name = maybeTableName;
+        } else {
+            if (attrs.tableArn) { throw new Error('Only one of tableArn or tableName can be provided'); }
+            name = attrs.tableName;
+            arn = stack.formatArn({
+                service: 'dynamodb',
+                resource: 'table',
+                resourceName: attrs.tableName,
+            });
+        }
+
+        return new Import(arn, name, attrs.tableStreamArn);
+    }
+
+    public readonly encryptionKey?: kms.IKey;
+
+    /**
+     * @attribute
+     */
+    public readonly tableArn: string;
+
+    /**
+     * @attribute
+     */
+    public readonly tableName: string;
+
+    /**
+     * @attribute
+     */
+    public readonly tableStreamArn: string | undefined;
+    public capacity?: Capacity
+
+    private readonly globalTable: CfnGlobalTable;
+
+    private readonly replicas = new Array<CfnGlobalTable.ReplicaSpecificationProperty>
+    private readonly replicaGlobalSecondaryIndexes = new Array<CfnGlobalTable.ReplicaGlobalSecondaryIndexSpecificationProperty>
+    private readonly keySchema = new Array<CfnTable.KeySchemaProperty>();
+    private readonly attributeDefinitions = new Array<CfnTable.AttributeDefinitionProperty>();
+    private readonly globalSecondaryIndexes = new Array<CfnGlobalTable.GlobalSecondaryIndexProperty>();
+    private readonly localSecondaryIndexes = new Array<CfnTable.LocalSecondaryIndexProperty>();
+
+    private readonly secondaryIndexSchemas = new Map<string, SchemaOptions>();
+    private readonly nonKeyAttributes = new Set<string>();
+
+    private readonly tablePartitionKey: Attribute;
+    private readonly tableSortKey?: Attribute;
+
+    private readonly billingMode: BillingMode;
+    private readonly tableScaling: ScalableAttributePair = {};
+    private readonly indexScaling = new Map<string, ScalableAttributePair>();
+    private readonly scalingRole: iam.IRole;
+
+    private readonly globalReplicaCustomResources = new Array<CustomResource>();
+
+    constructor(scope: Construct, id: string, props: TableProps) {
+        super(scope, id, {
+            physicalName: props.tableName,
+        });
+
+        this.billingMode = props.capacity?.billMode || BillingMode.PAY_PER_REQUEST
+
+        const { sseSpecification, encryptionKey } = this.parseEncryption(props);
+
+        let streamSpecification: CfnTable.StreamSpecificationProperty | undefined;
+        if (props.replicas) {
+            if (props.stream && props.stream !== StreamViewType.NEW_AND_OLD_IMAGES) {
+                throw new Error('`stream` must be set to `NEW_AND_OLD_IMAGES` when specifying `replicationRegions`');
+            }
+            streamSpecification = { streamViewType: StreamViewType.NEW_AND_OLD_IMAGES };
+
+            this.billingMode = props.billingMode ?? BillingMode.PAY_PER_REQUEST;
+        } else {
+            this.billingMode = props.billingMode ?? BillingMode.PAY_PER_REQUEST;
```

Comment:
This line occurs twice. It could occur only once, and doesn't have to be inside the `if (props.replicas)` check, right?